distribution/pull_v2.go

```diff
@@ -533,7 +533,8 @@ func (p *v2Puller) pullSchema2(ctx context.Context, ref reference.Named, mfst *s
 	}
 
 	configChan := make(chan []byte, 1)
-	errChan := make(chan error, 1)
+	errChan := make(chan error, 2)
+	downloadsDone := make(chan error, 1)
```

Comment:
This can be `downloadsDone := make(chan struct{})`, since no data is ever conveyed over it.